File: Abstraction/Java/Apartment.java
class Apartment extends Home implements Tenants{
	//Objective: Create a apartment class that is-a home
	private boolean patio; //Determines if the apartment has a patio
	
	public Apartment(){ //Default constructor for the Apartment class
		super(); //Calls the default constructor for the Home abstract class
		this.patio = false; //by default, instantiate the patio to false
	}
	public Apartment(int bedrooms, int bathrooms, boolean patio){
		super(bedrooms,bathrooms); //Calls the overloaded constructor for the Home abstract class
		this.patio = patio; //assigns value to patio
	}
	@Override
	public int Cost(){ //Abstract method declared in the abstract super class
		int rent = getBedrooms() * getBathrooms(); 
		//Calculates rent based on the number of bedrooms and bathrooms
		return rent;
	}
	@Override
	public void Welcome(){
		Integer tempBedrooms = new Integer(getBedrooms()); //Convert int to Integer
		Integer tempBathrooms = new Integer(getBathrooms()); //Convert int to Integer
		String bedrooms = Integer.toString(tempBedrooms); //Stores the string conversion of the integer
		String bathrooms = Integer.toString(tempBathrooms); //Stores the string conversion of the integer
		String statement = "Welcome to your "+bedrooms+" bedrooms " +bathrooms+" bathrooms home!"; //Stores the welcome
		System.out.println(statement);
	}
	@Override 
	public boolean getPatio(){ //Getter for patio field
		return patio;
	}
}

File: Abstraction/C++/Apartment.cpp
#include<iostream>
#include "Home.cpp"
using namespace std;

class Apartment : public Home{//Apartment is-a home
	private: //Apartment class fields
		bool patio; //initializes patio to false
	public: //Apartment class methods
		Apartment(){ //Apartment class default constructor
			Home(); //Call the default constructor of the super class
			patio = false; //initialize patio as falsed
		}
		Apartment(int bedrooms, int bathrooms, bool patio) : Home(bedrooms,bathrooms){ //Apartment overloaded constructor
			//Call the overloaded constructor of the super class
			this -> patio = patio;
		}
		int Cost(){ //Overrides the Cost() method from the super class
			int rent = getBedrooms() * getBathrooms(); //Calculate the cost of living
			return rent;
		}
};

//C++ multiple inheritance: class sub : public sup1, sup2,... calls the constructors in order

int main(){
	Apartment apartment(3,2,true);
	apartment.Welcome();
	return 0;
}
